code/modules/projectiles/guns/energy.dm

```diff
@@ -84,18 +84,7 @@
 	return new projectile_type(src)
 
 /obj/item/weapon/gun/energy/proc/get_external_cell()
-	if(isrobot(src.loc))
-		var/mob/living/silicon/robot/R = src.loc
-		return R.cell
-	if(istype(src.loc, /obj/item/rig_module))
-		var/obj/item/rig_module/module = src.loc
-		if(module.holder && module.holder.wearer)
-			var/mob/living/carbon/human/H = module.holder.wearer
-			if(istype(H) && H.back)
-				var/obj/item/weapon/rig/suit = H.back
-				if(istype(suit))
-					return suit.cell
-	return null
+	return loc.get_cell()
```

Comment:
I feel there are several ways this could be exploited

Reply:
Tell me some ways

Reply:
![image](https://user-images.githubusercontent.com/46600554/80899096-0f242980-8d36-11ea-818e-5b88918d20b1.png)

Reply:
It's mean that turfs can't return cell in get_cell() while it isn't coded